python/vote-text-bot-cloud-function/main.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2022 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START python-vote-text-bot]
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+import flask
+
+
+IMAGES = [
+    'https://media2.giphy.com/media/3oEjHK3aw2LcB1V3QQ/giphy.gif',
+    'https://media3.giphy.com/media/l0HlUIHlH4AKadXzy/giphy.gif',
+    'https://media0.giphy.com/media/3otPorfb8Lu7wjKllm/giphy.gif',
+    'https://media3.giphy.com/media/xT9IgFLBcm3Wi6l6qA/giphy.gif',
+]
+
+
+def vote(req: flask.Request) -> Mapping[str, Any]:
+  """ Google Cloud Function that responds to events sent from a Chat space.
+
+  Responds with the 'Vote' card together with a graphic.
+
+  Args:
+      req (flask.Request): the http request containing the chat message.
+
+  Returns:
+      Mapping[str, Any]: the response card
+  """
+  if req.method == 'GET':
+    return 'Sorry, this function must be called from a Google Chat.'
+
+  request = req.get_json(silent=True)
+  print(request)
+
+  match request.get('type'):
+    case 'ADDED_TO_SPACE' | 'MESSAGE':
```

Comment:
Might be better to have this triggered from @mention or slash command. I know it's a sample, but just feels weird to have a zero-context vote when something is added to a space :)

Shouldn't take much more code to trigger it with something like:

```
@vote Should we do X?
```